python/hail/utils/misc.py

```diff
@@ -260,12 +260,53 @@ def get_nice_field_error(obj, item):
 def check_collisions(fields, name, indices):
     from hail.expr.expressions import ExpressionException
     if name in fields and not fields[name]._indices == indices:
-        msg = 'name collision with field indexed by {}: {}'.format(list(fields[name]._indices.axes), name)
+        msg = "name collision with field indexed by {}: '{}'".format(list(fields[name]._indices.axes), name)
         error('Analysis exception: {}'.format(msg))
         raise ExpressionException(msg)
 
 def check_field_uniqueness(fields):
     for k, v in Counter(fields).items():
         if v > 1:
             from hail.expr.expressions import ExpressionException
-            raise ExpressionException("selection would produce duplicate field {}".format(repr(k)))
+            raise ExpressionException("selection would produce duplicate field '{}'".format(repr(k)))
+
+def check_keys(name, indices):
+    from hail.expr.expressions import ExpressionException
+    if name in set(indices.key):
+        msg = "cannot overwrite key field '{}' with annotate or select; use key_by to modify keys.".format(name)
```

Comment:
use just `{}` and `repr(name)`

This will escape unprintable characters appropriately